### PR TITLE
chore: add StepNavigation component to the lab package

### PR DESCRIPTION
**Pull request by plagoa**

### CODE REVIEW

File: packages/lab/src/components/StepNavigation/stepNavigationClasses.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  /** Styles applied to the li element. */
+  li?: string;
+  /**  Styles applied to the ol element. */
+  ol?: string;
+  /** Styles applied to the separator element. */
+  separator?: string;
+  /** Styles applied to the titles container element. */
+};
+
+const classKeys: string[] = ["root", "li", "ol", "separator"];
```
Comment (MEsteves22) [inline packages/lab/src/components/StepNavigation/stepNavigationClasses.ts:17]: `titles` is missing

File: packages/lab/src/components/StepNavigation/SimpleNavigation/Dot/Dot.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+              backgroundColor: getColor(state, theme),
+              width: dotSize,
+              height: dotSize,
+              "&:hover": {
+                backgroundColor: getColor(state, theme),
+              },
+            }),
+            state === "Current" && clsx(dotClasses.active, classes?.active),
+            (disabled ?? ["Current", "Disabled"].includes(state)) &&
+              clsx(dotClasses.ghostDisabled)
```
Comment (MEsteves22) [inline packages/lab/src/components/StepNavigation/SimpleNavigation/Dot/Dot.tsx:49]: Missing `classes.ghostDisabled`

File: packages/lab/src/components/StepNavigation/SimpleNavigation/Dot/dotClasses.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { getClasses } from "@hitachivantara/uikit-react-core";
+
+export type HvDotClasses = {
+  root: string;
+  active: string;
+  ghostDisabled: string;
```
Comment (MEsteves22) [inline packages/lab/src/components/StepNavigation/SimpleNavigation/Dot/dotClasses.ts:6]: The classes should be optional

File: packages/lab/src/components/StepNavigation/DefaultNavigation/Step/Step.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        )}
+        aria-label={`step-${title}`}
+        icon
+        variant="secondaryGhost"
+        overrideIconColors={false}
+        disabled={disabled ?? ["Current", "Disabled"].includes(state)}
+        onClick={onClick}
+      >
+        <StyledAvatar
+          className={clsx(stepClasses.avatar, classes?.avatar, classes?.[size])}
```
Comment (MEsteves22) [inline packages/lab/src/components/StepNavigation/DefaultNavigation/Step/Step.tsx:115]: Missing `stepClasses[size]`

**Review by MEsteves22 — APPROVED** [review]

**Review by HQFOX — APPROVED** [review]

**Pull request merged**